src/utils/solanaUtils.ts

```diff
@@ -1091,6 +1115,94 @@ export async function getCachedKeyToAssets(
     .filter(truthy)
 }
 
+export async function getCachedIotInfo(
+  hemProgram: Program<HeliumEntityManager>,
+  infoKey: PublicKey,
+): Promise<IotHotspotInfoV0 | undefined> {
+  const cache = await getSingleton(hemProgram.provider.connection)
+  const iotInfo = await cache.search(
+    infoKey,
+    (pubkey, account) => ({
+      pubkey,
+      account,
+      info: hemProgram.coder.accounts.decode<
+        IdlAccounts<HeliumEntityManager>['iotHotspotInfoV0']
+      >('IotHotspotInfoV0', account.data),
+    }),
+    true,
+    false,
+  )
+
+  return iotInfo?.info
+}
+
+export async function getCachedIotInfos(
+  hemProgram: Program<HeliumEntityManager>,
+  infoKeys: PublicKey[],
+): Promise<IotHotspotInfoV0[]> {
+  const cache = await getSingleton(hemProgram.provider.connection)
+  return (
+    await cache.searchMultiple(
+      infoKeys,
+      (pubkey, account) => ({
+        pubkey,
+        account,
+        info: hemProgram.coder.accounts.decode<
+          IdlAccounts<HeliumEntityManager>['iotHotspotInfoV0']
+        >('IotHotspotInfoV0', account.data),
+      }),
+      true,
```

Comment:
This should be false, as iot infos are not static (they can change, for example in an assert location). Setting this to true will tell the cache to not watch for changes, which could cause issues updating.